docs/changelog/96161.yaml

```diff
@@ -4,3 +4,21 @@ area: "Search"
 type: enhancement
 issues:
  - 95541
+highlight:
+  title: Better indexing and search performance under concurrent indexing and search
+  body: "When a query like a match phrase query or a terms query targets a constant keyword field \
+  we can skip the query execution on shards where the query is rewritten to match no documents.\
+  We take advantage of index mappings including constant keyword fields and rewrite queries in such a way that, if\
+  a constant keyword field does not match a value defined in the index mapping we rewrite the query to match no\
+  document. This will result in the shard level request to return immediately, before the query is executed on the data\
+  node and, as a result, skipping the shard completely. In a real world scenario it is likely that index patterns or\
+  data streams include tens or hundreds of backing indices each of them with multiple shards involved. Skipping shards\
+  in such scenario might result in better query performance and better cluster resource usage. Note, anyway, that\
```

Comment:
This exlpains how skipping shards helps with search performance, which is true, but Elasticsearch has had this for years. Let's focus instead of the new change, which is that the `can_match` phase no longer needs to refresh search-idle shards in a number of cases?

Quickly mention better indexing performance here as well thanks to less refreshing and merging?